### PR TITLE
python3Packages.types-redis: fix build and add pyopenssl

### PR DESCRIPTION
**Pull request by gador**
###### Description of changes

`types-redis` requires `pyopenssl`. This is now added.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 221974` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
8 packages built:
</summary>
<ul>
<li>python310Packages.types-pyopenssl</li>
<li>python310Packages.types-pyopenssl.dist</li>
<li>python310Packages.types-redis</li>
<li>python310Packages.types-redis.dist</li>
<li>python311Packages.types-pyopenssl</li>
<li>python311Packages.types-pyopenssl.dist</li>
<li>python311Packages.types-redis</li>
<li>python311Packages.types-redis.dist</li>
</ul>
</details>

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 221974` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
8 packages built:
</summary>
<ul>
<li>python310Packages.types-pyopenssl</li>
<li>python310Packages.types-pyopenssl.dist</li>
<li>python310Packages.types-redis</li>
<li>python310Packages.types-redis.dist</li>
<li>python311Packages.types-pyopenssl</li>
<li>python311Packages.types-pyopenssl.dist</li>
<li>python311Packages.types-redis</li>
<li>python311Packages.types-redis.dist</li>
</ul>
</details>

File: pkgs/development/python-modules/types-pyopenssl/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, cryptography
+}:
+
+buildPythonPackage rec {
+  pname = "types-pyOpenSSL";
```
Comment (fabaff) [inline pkgs/development/python-modules/types-pyopenssl/default.nix:8]: `pname` should be normalized according to PEP-503, see also https://nixos.org/manual/nixpkgs/stable/#contributing

```suggestion
  pname = "types-pyopenssl";
```

File: pkgs/development/python-modules/types-pyopenssl/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  version = "23.0.0.4";
+  format = "setuptools";
+
+  src = fetchPypi {
+    inherit pname version;
```
Comment (fabaff) [inline pkgs/development/python-modules/types-pyopenssl/default.nix:13]:
```suggestion
    pname = "types-pyOpenSSL";
    inherit version;
```

File: pkgs/development/python-modules/types-pyopenssl/default.nix (hunk 1 above)
Comment (gador) [reply pkgs/development/python-modules/types-pyopenssl/default.nix:8]: it doesn't find the package on pypi when normalized

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 221974` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
8 packages built:
</summary>
<ul>
<li>python310Packages.types-pyopenssl</li>
<li>python310Packages.types-pyopenssl.dist</li>
<li>python310Packages.types-redis</li>
<li>python310Packages.types-redis.dist</li>
<li>python311Packages.types-pyopenssl</li>
<li>python311Packages.types-pyopenssl.dist</li>
<li>python311Packages.types-redis</li>
<li>python311Packages.types-redis.dist</li>
</ul>
</details>

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 221974` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
8 packages built:
</summary>
<ul>
<li>python310Packages.types-pyopenssl</li>
<li>python310Packages.types-pyopenssl.dist</li>
<li>python310Packages.types-redis</li>
<li>python310Packages.types-redis.dist</li>
<li>python311Packages.types-pyopenssl</li>
<li>python311Packages.types-pyopenssl.dist</li>
<li>python311Packages.types-redis</li>
<li>python311Packages.types-redis.dist</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
@ofborg build python310Packages.types-redis python310Packages.types-pyopenssl
